### ISSUE TITLE
Find by type appears to be somewhat affected by type var names chosen

### ISSUE THREAD

**Issue by alvaroc1** [unisonweb/unison#1015]
```
.base.List> find : (b -> a -> b) -> b -> [a] -> b
  1. foldLeft : (b ->{𝕖} a ->{𝕖} b) -> b -> [a] ->{𝕖} b
.base.List> find : (a -> b -> a) -> a -> [b] -> a
  :point_up:
  I couldn't find exact type matches, resorting to fuzzy matching...
  :no_mouth:
  No results. Check your spelling, or try using tab completion to supply command arguments.
.base.List>
```

**Comment by aryairani** [comment]
> It's probably just iterating over all the free variables in a `Set`, and constructing `forall` in that order, which will yield a different hash based on the sort order of the variable names relative to where they're used.

e.g. if `a -> b` becomes `forall a b. a -> b` then `b -> a` should become `forall b a. b -> a`, but it becomes `forall a b. b -> a` instead, which has a different hash.

**Comment by pchiusano** [comment]
Yeah, we could sort the type variables based on the place in the type where they are used, or something similarly deterministic. Could also change the hashing to do this normalization but that would require a repoformat upgrade.

Could also just do the same search with multiple instantiations of the variable ordering.

**Comment by pete-ts** [comment]
I have investigated and found that the current behaviour is along the lines of what @aryairani described, the issue being in 

https://github.com/unisonweb/unison/blob/f7e36e7583a130582430ecf47b5a3e7cfc1d7b8f/unison-core/src/Unison/Type.hs#L503-L513

`Type.hs: generalizeLowercase`, the `set` operation used does not maintain the order of the arguments in the `t` being passed.

I think I have a partial fix for this, but in the process I seem to have broken a unittest (no idea how or where to fix that). 

Can raise a PR as long as someone has some time to help guide me to the complete solution?
